controllers/user.controller.js

```diff
@@ -0,0 +1,61 @@
+const { CastError } = require("mongoose");
+const bcrypt = require("bcryptjs");
+const User = require("../models/user.schema");
+const { postUserSchema, patchUserSchema } = require("../validation/user.validation");
+const logger = require("../config/logger");
+
+// user create
+const createUser = async (req, res) => {
+  try {
+    const { error, value } = postUserSchema.validate(req.body);
+    if (error) {
+      return res.status(400).send({ message: error.message });
+    }
+    value.password = await bcrypt.hash(value.password, 10);
+    const user = new User(value);
+    const userEmail = await User.findOne({ email: req.body.email });
+    if (userEmail) {
+      res.status(409).send({ message: "Email already registered" });
+    } else {
+      await user.save();
+      res.status(201).send(user);
+    }
+  } catch (error) {
+    if (error instanceof CastError) {
+      res.status(400).send({ message: "Invalid User Details" });
+    } else {
+      logger.error("Could not create user: ", error);
+      res.status(500).send({ message: "Internal Server Error" });
+    }
+  }
+  return null;
+};
+
+const updateUser = async (req, res) => {
+  try {
```

Comment:
- We need to validate that user is logged using Authorization header - https://stackabuse.com/authentication-and-authorization-with-jwts-in-express-js/ Check section `At this point, let's create the Express middle`. For this you need to understand https://expressjs.com/en/guide/using-middleware.html
- And the Logged in user's id and request.params.id should be same otherwise return `Unauthorized`,This can be also done through a middleware